Zone5/Views/ThirdPartyConnectionsView.swift

```diff
@@ -48,7 +48,16 @@ public class ThirdPartyConnectionsView: APIView {
 	@discardableResult
 	public func setThirdPartyToken(type: UserConnectionType, parameters: URLEncodedBody?, completion: @escaping Zone5.ResultHandler<Zone5.VoidReply>) -> PendingRequest? {
 		let endpoint = Endpoints.confirmConnection.replacingTokens(["connectionType": type.connectionName])
-		return get(endpoint, parameters: parameters, expectedType: Zone5.VoidReply.self, with: completion)
+		
+		var queryItems: [URLQueryItem] = []
+		
+		if let parameters = parameters {
+			queryItems = parameters.queryItems
+			queryItems.append(URLQueryItem(name: "noredirect", value: "true"))
+		}
```

Comment:
parameters is defined as optional and then you are only appending noredirect if it is not nil. Technically this allows for noredirect to be omitted, which is not correct.

There are 2 solutions which are valid:
1) Make parameters not optional
2) always append noredirect (move line 56 outside of the if statement)

In my opinion making parameters not optional is the best approach as I can't think of a valid scenario where it would be nil.

Reply:
Agreed. That said, optional `parameters` does appear to be in line with the rest of the code base (in particular, the `View.get(_:parameters:expectedType:with:)` method that is called here). Either way, `parameters` being `nil` is equivalent to an empty parameter set, and the redirect flag should be set in both cases.

Additionally, there should be unit tests to cover the expectations for when `parameters` is empty and/or `nil`.